File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    pdic = {}
    for _ in participant :
        if _ in pdic:
            pdic[_] += 1
        else :
            pdic[_] = 1

    for _ in completion:
        pdic[_] += -1

    for _ in participant:
        if pdic[_] != 0 :
            return _

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String,Integer> map = new HashMap<>();
        for(int i=0; i<participant.length; i++){
            map.put(participant[i], map.getOrDefault(participant[i], 0) + 1);
        }
        
        for(int i=0; i<completion.length; i++){
            map.put(completion[i], map.get(completion[i]) -1);
        }
        
        for(int i=0; i<participant.length; i++){
            if(map.get(participant[i]) > 0) answer = participant[i];
        }
        return answer;
    }
}